spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java

```diff
@@ -135,10 +140,23 @@ protected TableCatalog checkSourceCatalog(CatalogPlugin catalog) {
 
   @Override
   public SnapshotAction withLocation(String location) {
-    Preconditions.checkArgument(!sourceTableLocation().equals(location),
+    String qualifiedLocation = makeQualifiedLocation(location);
+    Preconditions.checkArgument(!sourceTableLocation().equals(qualifiedLocation),
         "Cannot create snapshot where destination location is the same as the source location." +
             " This would cause a mixing of original table created and snapshot created files.");
     this.destTableLocation = location;
     return this;
   }
+
+  private String makeQualifiedLocation(String location) {
+    try {
+      Path path = new Path(location);
+      Configuration conf = spark().sessionState().newHadoopConf();
+      FileSystem fs = FileSystem.get(path.toUri(), conf);
+      return CatalogUtils.URIToString(fs.makeQualified(path).toUri());
+    } catch (IOException ioe) {
+      LOG.error("Failed to initialize hadoop filesystem.", ioe);
+      return location;
```

Comment:
If failed, back to location. Any suggestions?

Reply:
We should probably just fail, after all if the location string cannot be resolved isn't it going to fail anyway?

Reply:
make sense @RussellSpitzer